src/software/uart/boost_uart_communication.cpp

```diff
@@ -4,10 +4,10 @@
 #include <boost/asio.hpp>
 #include <iostream>
 
-BoostUartCommunication::BoostUartCommunication(IoService& io_service, int baud_rate,
+BoostUartCommunication::BoostUartCommunication(int baud_rate,
                                                std::string device_serial_port)
 {
-    openPort(io_service, baud_rate, device_serial_port);
+    openPort(io_service_, baud_rate, device_serial_port);
 }
```

Comment:
io_service should probably be explicitly initialized

Reply:
done